Sources/Orbit/Components/Slider.swift

```diff
@@ -0,0 +1,421 @@
+    }
+
+    private var accessibilityHint: String {
+        startLabel + "—" + endLabel
+    }
+
+    private func updateSelectedRangeIfNeeded() {
+        if let upperBound = secondValue, case let .range(selectedRange) = value {
+            selectedRange?.wrappedValue = min(firstValue, upperBound)...max(firstValue, upperBound)
+        }
+    }
+
+    private func currentValue(location: CGFloat, sliderWidth: CGFloat) -> Value {
+        return bounds.lowerBound + bounds.upperBound * Value(location / sliderWidth)
+    }
+
+    private func position(by value: Value, sliderWidth: CGFloat) -> CGFloat {
+        let position = CGFloat(value / bounds.upperBound) * sliderWidth
+        let normalisied = min(sliderWidth, max(0, position))
+        return normalisied
+    }
+
+    public init(
+        value: Binding<Value>,
+        valueLabel: String,
+        endLabel: String,
+        bounds: ClosedRange<Value>,
+        step: Value.Stride? = nil
+    ) {
+        self.value = .single(value)
+        self.valueLabel = valueLabel
+        self.startLabel = ""
+        self.endLabel = endLabel
+
+        self.bounds = bounds
+        self.step = step
+
+        self._firstValue = State(initialValue: value.wrappedValue)
+        buckets = bounds.chunked(into: step ?? Value.Stride(bounds.upperBound))
+    }
+
+    public init(
+        selectedRange: Binding<ClosedRange<Value>>,
+        valueLabel: String,
+        startLabel: String,
+        endLabel: String,
+        bounds: ClosedRange<Value>,
+        step: Value.Stride? = nil
+    ) {
+        assert(step != 0, "step can'be 0")
+
+        self.value = .range(selectedRange)
+        self.valueLabel = valueLabel
+        self.startLabel = startLabel
+        self.endLabel = endLabel
+        self.bounds = bounds
+        self.step = step
+
+        self._firstValue = State(initialValue: selectedRange.wrappedValue.lowerBound)
+        self._secondValue = State(initialValue: selectedRange.wrappedValue.upperBound)
+
+        self.buckets = bounds.chunked(into: step ?? Value.Stride(bounds.upperBound))
+    }
+}
+
+struct SliderThumb: View {
+
+    @Environment(\.sizeCategory) var sizeCategory
+
+    public static let circleDiameter: CGFloat = 24
+    public static let dotDiameter: CGFloat = 8
+
+    public var body: some View {
+        Circle()
+            .frame(width: circleDiameter, height: circleDiameter)
+            .foregroundColor(Color.white)
+            .elevation(.level1)
+            .overlay(indicatorSymbol)
+    }
+
+    @ViewBuilder var indicatorSymbol: some View {
+        Circle()
+            .foregroundColor(Color.blueNormal)
+            .frame(width: dotDiameter, height: dotDiameter)
+    }
+
+    var circleDiameter: CGFloat {
+        Self.circleDiameter * sizeCategory.controlRatio
+    }
+
+    var dotDiameter: CGFloat {
+        Self.dotDiameter * sizeCategory.ratio
+    }
+}
+
+extension ClosedRange where Bound: BinaryFloatingPoint {
```

Comment:
`extension ClosedRange<BinaryFloatingPoint>`

Reply:
Is it really the same? In case of you case compiler complains about `Binary operator '-' cannot be applied to two 'Bound' operands`